pkg/webhook/ingress/types.go

```diff
@@ -44,6 +47,24 @@ func (n NetworkingV1) IngressClass() (res *string) {
 	return
 }
 
+func (n NetworkingV1) SetIngressClass(ingressClassName string) {
+	if n.Spec.IngressClassName == nil {
+		if a := n.GetAnnotations(); a != nil {
+			if _, ok := a[annotationName]; ok {
```

Comment:
Do we need this check? `SetIngressClass` is called by the mutating webhook, accepting as its argument the default ingress class, we don't need to check if the object has already it.

Reply:
No, we don't.